### PR TITLE
Add support for exporting GPT-J to ONNX-TRT

### LINKED ISSUE #15640: Add support for ONNX-TensorRT conversion for GPT-J6B (and possible bug in rotary embedding)
### Who can help
@patil-suraj 

## Information

Model I am using: GPT-J

The problem arises when using:
* [x] the official example scripts: (give details below)
* [x] my own modified scripts: (give details below)

## Description
I opened this issue for two reasons:
1. This is not strictly a bug report, rather a change that enables converting this model to ONNX and then parsing it using the current TensorRT ONNX parser.
2. Possible implementation bug in GPT-J.

## Details
1. When exporting GPT-J to ONNX using the latest version (v4.16.2), one of the ops that is exported is [SplitToSequence](https://github.com/onnx/onnx/blob/main/docs/Operators.md#SplitToSequence) (along with more Sequence* ops) that is currently not supported in the [TensorRT ONNX parser](https://github.com/onnx/onnx-tensorrt/blob/master/docs/operators.md).
This is entirely due to just 1 line of code that uses `torch.repeat_interleave`. ([relevant line](https://github.com/huggingface/transformers/blob/52d2e6f6e904ef9b75c78716ce77b98196ed837a/src/transformers/models/gptj/modeling_gptj.py#L67))
```
sin, cos = map(lambda t: t[None, offset : x.shape[1] + offset, None, :].repeat_interleave(2, 3), sincos)
```
By replacing `lambda t` with this:
```
lambda t: t.view(-1, 1).repeat(1, 2).view(seq_len, -1)[None, offset : x.shape[1] + offset, None, :]
```
we get the exact same output tensors but now exporting to ONNX doesn't include any Sequence* ops, and TensorRT can parse it successfully.
The suggested function is even faster, although probably not critical in this huge model (benched only on CPU):
```
original: 106 µs ± 20.9 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
suggested: 32.4 µs ± 6.55 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)
```

2. I was following the implementation in EleutherAI for rotary positional embeddings and I'm trying to understand if this is a bug or I'm simply missing something (would love an explanation if you can spare the time) but there (EleutherAI) they implement this function (rotary positional embedding) using `torch.cat` instead of `torch.repeat_interleave`, as can be seen [here](https://github.com/EleutherAI/gpt-neox/blob/b30afd1d0a1d06220be9b5f2c9c9c1523defba96/megatron/model/positional_embeddings.py#L41).

If I'm not missing something, the EleutherAI version transforms a tensor from
```
[[1,2,3],
 [4,5,6]]
```
to 
```
[[1,2,3,1,2,3],
 [4,5,6,4,5,6]]
```
and HF version (using repeat_interleave):
```
[[1,2,3],
 [4,5,6]]
```
to 
```
[[1,1,2,2,3,3],
 [4,4,5,5,6,6]]
```
Can anyone confirm the current implementation is indeed correct? Because otherwise `cat` and `repeat_interleave` are very different, and the rest of the implementation doesn't take it into account.

### PR DESCRIPTION
**Pull request by tomerip**
This change enables exporting GPT-J to ONNX without having any `Sequence*` ops which are currently unsupported in TensorRT ONNX parser. It works by using a sequence of "`view`, `repeat`, `view`" instead of using `repeat_interleave`.

# What does this PR do?
Fixes #15640 

## Who can review?
@patil-suraj

### CODE REVIEW

File: src/transformers/models/gptj/modeling_gptj.py [hunk 1]
```diff
@@ -63,7 +63,9 @@ def rotate_every_two(x):
 
 
 def apply_rotary_pos_emb(x, sincos, offset=0):
-    sin, cos = map(lambda t: t[None, offset : x.shape[1] + offset, None, :].repeat_interleave(2, 3), sincos)
+    sin, cos = map(
+        lambda t: t.view(-1, 1).repeat(1, 2).view(t.shape[0], -1)[None, offset : x.shape[1] + offset, None, :], sincos
+    )
```
Comment (patil-suraj) [inline src/transformers/models/gptj/modeling_gptj.py:68]: Nice!

(nit) this looks a bit too magical, maybe split this on multiple lines and add a comment explaining what it's doing.

**Review by patil-suraj — APPROVED** [review]
Thank you for working on this!

**Review by lewtun — APPROVED** [review]
Thank you adding this fix @tomerip 🔥 ! Looks good to me with @patil-suraj's suggestion to make the change a bit easier to reason about

**Comment by tomerip** [comment]
Added a function to handle the repeat, commenting on each step.

**Pull request merged**